ambry-router/src/main/java/com.github.ambry.router/GetBlobInfoOperation.java

```diff
@@ -323,7 +323,11 @@ private void checkAndMaybeComplete() {
 
     if (operationCompleted) {
       Exception e = operationException.get();
+      if (operationResult == null && e == null) {
+        e = new RouterException("Operation failed, but exception was not set", RouterErrorCode.UnexpectedInternalError);
```

Comment:
The 'UnexpectedInternalError' metric gets updated in this flow, which is sufficient.
This shouldn't ideally happen, this is just a guard as mentioned in the description.